### DISCUSSION TITLE
Duda con complejidad en parte 2

### DISCUSSION THREAD

**Discussion by gaspar-mr05** [IIC2133-PUC/2024-1#73] [Tarea 1] [answered]
Hola. En la parte 2 de la tarea nos piden programar los eventos con alguna complejidad en especifico? (como en la parte 1). Lo digo porque en la parte 1 hice balanceo de los arboles precisamente para tener la complejidad pedida, pero en la parte 2 pienso no hacerlo ya que en ningun lado piden complejidad (por lo menos en el enunciado)

**Comment by Segismundios** [comment]
Holaa, no soy ayudante, pero creo que esta parte del enunciado implica que sí se espera cierta complejidad al manejar los eventos de la parte 2:
"Cabe destacar que este problema de búsqueda debe cumplir con los principios de eficiencia y optimización
aprendidos en clases, garantizando que el proceso de búsqueda sea lo más rápido y preciso posible, utilizando
algoritmos y estructuras de datos avanzadas para manejar grandes volúmenes de datos con facilidad y precisión."

**Comment by sachondo7** [comment, marked as answer]
Ahí tienes que decidir tú que tipo de edds son las adecuadas para la parte 2. El compañero de arriba tiene razón con lo que dice, es esperable que usen técnicas para manejar con hartos datos. Como dice el enunciado: 
**Hint: Puedes usar distintos tipos de árboles para cada una de las búsquedas. Investiga de como se puede
trabajar con múltiples valores en los árboles de búsqueda. Aunque toda esta parte se puede resolver con los
árboles vistos en clases, puede ser útil investigar otro tipo de árboles de búsqueda, como RangeTree.
Importante: Deberán usar estructuras de datos con complejidad de búsqueda promedio a lo más O(log(n)).**
